src/app/dashboard/dashboard.component.ts

```diff
@@ -4,19 +4,94 @@ import { UserService } from '../shared/user.service';
 
 // Main page once logged in.  At this stage is more of a placeholder.
 @Component({
-  template: `
-    <div id="greeting">Hi, {{name}}</div>
-  `
+  templateUrl: './dashboard.component.html',
+  styleUrls: [ './dashboard.scss' ]
 })
-export class DashboardComponent implements OnInit {
-  name = '';
-  roles: string[] = [];
 
+export class DashboardComponent implements OnInit {
+  myLibrary = [
+    { text: 'library one' },
+    { text: 'library two' },
+    { text: 'library three' },
+    { text: 'library four' },
+    { text: 'library five' },
+    { text: 'library six' },
+    { text: 'library seven' },
+    { text: 'library Eight' },
+    { text: 'library nine' },
+    { text: 'library ten' },
+  ];
+  myTeam = [
+    { text: 'team one' },
+    { text: 'team two' },
+    { text: 'team three' },
+    { text: 'team four' },
+    { text: 'team five' },
+    { text: 'team six' },
+  ];
+  myMeetups = [
+    { text: 'meetup one' },
+    { text: 'meetup two' },
+    { text: 'meetup three' },
+    { text: 'meetup four' },
+    { text: 'meetup five' },
+    { text: 'meetup six' },
+    { text: 'meetup seven' },
+  ];
+  myCourses = [
+  { text: 'courses one' },
+  { text: 'courses two' },
+  { text: 'courses three' },
+  { text: 'courses four' },
+  { text: 'courses five' },
+  { text: 'courses six' },
+  { text: 'courses seven' },
+  ];
   constructor(
     private userService: UserService
   ) {}
 
   ngOnInit() {
     Object.assign(this, this.userService.get());
   }
+
+  nextLibrary() {
+    this.myLibrary = this.myLibrary.concat( this.myLibrary.splice( 0, 4 ) );

```

Comment:
instead of playing around with array add pointer to point out the current position